eng/pipelines/templates/stages/archetype-python-release.yml

```diff
@@ -28,6 +28,13 @@ stages:
                 deploy:
                   steps:
                     - checkout: self
+                    
+                    - task: UsePythonVersion@0
```

Comment:
This dependency is not intuitive based on calling the verify-changelog.yml. IIRC we essentially need to do this as a pre-step anytime we use the shared powershell scripts. So I'd suggest we either do this as part of the script call to ensure it is always done or we eliminate the need to call scripts. From the python repo in our eng/common scripts. 

@chidozieononiwu I'm starting to wonder if we should move the scripts for parsing out the language specific metadata into the language repos and just have an extension point in our common scripts. Perhaps instead of passing the language we pass a class or a script file for the particular language.

Reply:
@weshaggard I will revisit this one one release is completed for synapse. I think we need to look at broader point of view as you suggested.

Reply:
OK if this is just to unblock the current releases then please move forward with it. I thought they were unblocked with your earlier change and this was the follow-up.